src/main/host/descriptor/tcp.c

```diff
@@ -2336,7 +2336,7 @@ static gssize _tcp_sendUserData(Transport* transport, Thread* thread, PluginVirt
     gsize remaining = MIN(acceptable, space);
 
     /* break data into segments and send each in a packet */
-    gsize maxPacketLength = CONFIG_MTU - CONFIG_HEADER_SIZE_TCPIP - CONFIG_HEADER_SIZE_ETH;
+    gsize maxPacketLength = CONFIG_MTU - CONFIG_HEADER_SIZE_TCPIP;
```

Comment:
Consider using the `CONFIG_TCP_MAX_SEGMENT_SIZE` constant defined in `src/main/core/support/definitions.h`.